docs/src/testing/storage-cheatcodes.md

```diff
@@ -0,0 +1,164 @@
+which allow manipulating the storage directly (reading and writing).
+
+In order to obtain the variable address that you'd like to write to, or read from, you need to use either:
+- `selector!` macro - if the variable is not a mapping
+- `map_entry_address` function in tandem with `selector!` - for key-value pair of a map variable
+
+## Example: Felt-only storage
+This example uses only felts for simplicity
+
+```rust
+#[starknet::contract]
+mod Contract {
+    #[storage]
+    struct Storage {
+        plain_felt: felt252,
+        mapping: LegacyMap<felt252, felt252>,
+    }
+}
+
+// ...
+use snforge_std::{ store, load, map_entry_address };
+
+#[test]
+fn store_and_load_with_plain_felt() {
+    // ...
+    store(contract_address, selector!("plain_felt"), array![123].span());
+    // plain_felt = 123
+    let loaded = load(contract_address, selector!("plain_felt"), 1);
+    assert(loaded.len() == 1, 'Wrong loaded vector');
+    assert(*loaded.at(0) == 123, 'Wrong loaded value');
+}
+
+
+#[test]
+fn store_and_load_map_entry() {
+    // ...
+    store(
+        contract_address, 
+        map_entry_address(
+            selector!("mapping"), // Providing variable name
+            array![123].span(),   // Providing mapping key 
+        ),
+        array![321].span()
+    );
+    
+    // mapping = { 123: 321, ... }
+    let loaded = load(
+        contract_address, 
+        map_entry_address(
+            selector!("mapping"), // Providing variable name
+            array![123].span(),   // Providing mapping key 
+        ),
+        1,
+    );
+    
+    assert(loaded.len() == 1, 'Expected 1 felt loaded');
+    assert(*loaded.at(0) == 321, 'Expected 321 value loaded');
+}
+```
+
+## Example: Complex structures in storage
+This example uses a complex key and value, with `serde` as default serialization method.
+
+```rust
+use snforge_std::{ store, load, map_entry_address };
+
+#[starknet::contract]
+mod Contract {
+    #[derive(Serde)]
+    struct MapKey {
+        a: felt252,
+        b: felt252,
+    }
+
+    // Required for indexing of complex_mapping
+    impl StructuredKeyHash of LegacyHash<MapKey> {
+        fn hash(state: felt252, value: MapKey) -> felt252 {
+            let state = LegacyHash::<felt252>::hash(state, value.a);
+            LegacyHash::<felt252>::hash(state, value.b)
+        }
+    }
+
+    #[derive(Serde, starknet::Store)]
+    struct MapValue {
+        a: felt252,
+        b: felt252,
+    }
+    
+    // Serialization of key with `Serde` (other methods are possible)
+    impl MapKeyIntoSpan of Into<MapKey, Span<felt252>> {
+        fn into(self: MapKey) -> Span<felt252> {
+            let mut serialized_struct: Array<felt252> = array![];
+            self.serialize(ref serialized_struct);
+            serialized_struct.span()
+        }
+    }
+    
+     // Serialization of value with `Serde` (other methods are possible)
```

Comment:
The description may be misleading - it seems like this implementation somehow affects the `Store` trait while it is just to make using cheatcodes easier. We should state the purpose of the code